app/App.hs

```diff
@@ -32,7 +32,9 @@ data App :: Effect where
   AskGlobalOptions :: App m GlobalOptions
   FromAppPathFile :: AppPath File -> App m (Path Abs File)
   GetMainAppFile :: Maybe (AppPath File) -> App m (AppPath File)
+  GetMainAppFileMaybe :: Maybe (AppPath File) -> App m (Maybe (AppPath File))
   GetMainFile :: Maybe (AppPath File) -> App m (Path Abs File)
+  GetMainFileMaybe :: Maybe (AppPath File) -> App m (Maybe (Path Abs File))
```

Comment:
Q: Do `GetMain{File, AppFile}` need to be defined on the effect as they can be derived from `GetMain{File, AppFile}Maybe?

Reply:
I'll improve the app effect in a separate pr